resources/js/beatmap-discussions/beatmap-owner-editor.tsx

```diff
@@ -206,70 +185,61 @@ export default class BeatmapOwnerEditor extends React.Component<Props> {
     );
   }
 
-  private renderUsername() {
-    if (!this.editing) {
-      return (
-        <UserLink
-          className='beatmap-owner-editor__input beatmap-owner-editor__input--static'
-          user={this.props.user}
-        />
-      );
-    }
+  private readonly renderOwner = (owner: UserJson, onRemoveClick: (user: UserJson) => void) => (
+    <BeatmapOwner key={owner.id} editing={this.editing} onRemoveUser={onRemoveClick} user={owner} />
+  );
 
+  private renderUsernames() {
     return (
-      <input
-        ref={this.inputRef}
-        className={classWithModifiers('beatmap-owner-editor__input', {
-          error: this.inputUser == null,
-        })}
-        disabled={this.updatingOwner}
-        onChange={this.handleUsernameInput}
-        onKeyUp={this.handleUsernameInputKeyup}
-        value={this.inputUsername}
-      />
+      <InputContainer
+        for='beatmap-owner-editor-username-input'
+        hasError={!this.canSave}
+        modifiers='beatmap-owner-editor'
+        showError={this.showError}
+      >
+        <div className={classWithModifiers('beatmap-owner-editor-owners', { editing: this.editing })}>
+          {this.editing ? (
+            <UsernameInput
+              id='beatmap-owner-editor-username-input'
```

Comment:
should have unique id, otherwise clicking on any entry label will always focus the first box

Reply:
apparently if you render the input box first and move it to the end using flex's `order: 1`, the id isn't needed